src/qililab/transpiler/transpiler.py

```diff
@@ -68,6 +86,22 @@ def optimize_transpilation(nqubits: int, ngates: list[gates.Gate]) -> list[gates
             raise NotImplementedError(f"{gate.name} not part of supported gates {supported_gates}")
         if isinstance(gate, gates.RZ):
             shift[gate.qubits[0]] += gate.parameters[0]
+        # add CZ phase correction
+        elif isinstance(gate, gates.CZ) and gates_settings is not None:
+            gate_settings = gates_settings.get_gate(name=gate.__class__.__name__, qubits=gate.qubits)
+            control_qubit, target_qubit = gate.qubits
+            corrections = next(
+                (
+                    event.pulse.options
+                    for event in gate_settings
+                    if f"q{control_qubit}_phase_correction" in event.pulse.options
```

Comment:
This is the problem of using generic kwargs...

Why did we choose this option over adding this the phase shift option in the dataclass?

Reply:
it is to keep `gate_settings` general and avoid having a part of the code specifically handling `CZ` gates. I wanted to avoid discriminating between gates because this is what happened when we had `harware_gates` which ended up making gate definitions too rigid.  Perhaps it'd be nice to discuss this in code discussions

Reply:
Why not adding a phase correction as a generic feature in the gate transpilation? Meaning it can be added for any gate the user defines.

Reply:
Something like the following:

```yml
CZ(0,2):
- bus: flux_line_q2_bus
  pulse:
    amplitude: 1.0
    phase: 0
    duration: 101
    shape:
      name: snz
      t_phi: 1
      b: 0.5
    phase_correction:
      - qubit: 0
         phase: 0.1
       - qubit: 1
         phase: 0.4
```

Reply:
Where the `phase_correction` can be added to any gate/pulse

Reply:
phase correction is specific to 2q gates, and specifically for the qubit gates we're implementing (based on interactions with the qubits' avoided crossing). So it doesn't make a lot of sense to add a phase correction for other gates.
With the generic "options" i wanted to avoid both this:
> it is to keep gate_settings general and avoid having a part of the code specifically handling CZ gates. I wanted to avoid discriminating between gates because this is what happened when we had harware_gates which ended up making gate definitions too rigid. Perhaps it'd be nice to discuss this in code discussions
mentioned above but also the opposite: adding too many parameters to gate_settings which in many cases might be empty.

Personally I don't think it is a problem to allow gate definitions to be as flexible as possible - but I do agree that the `options` method I propose here is uncertain so we'd need documentation to inform and be clear of the possible options.
I envision these `options` parameter as a similar case as options in [`scipy.minimize`](https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.minimize.html) or the config options in[ dwave's sampler method](https://github.com/dwavesystems/dwave-system/blob/1.20.0/dwave/system/samplers/dwave_sampler.py#L91) in both cases, there's an arbitrary dictionary of options (because they are method dependent) but this is not a problem because there's a lot of documentation supporting it (more so in scipy's case)

Reply:
I agree! Let's make sure we add documentation for this then!

I believe it makes sense to mention this in the Runcard page of the documentation. Could you create a task in the backlog to make sure this is mentioned?

Reply:
If we do this, then I have no problem merging this PR